core/templates/dev/head/components/HintEditorDirective.js

```diff
@@ -55,7 +55,14 @@ oppia.directive('hintEditor', [
 
           $scope.saveThisHint = function() {
             $scope.hintEditorIsOpen = false;
+            var contentHasChanged = 
+              $scope.hintMemento.hintContent.getHtml() !=
```

Comment:
Put parens around this and use !==

```
             var contentHasChanged = (
               $scope.hintMemento.hintContent.getHtml() !==
               $scope.hint.hintContent.getHtml());
```

Ditto elsewhere.

Reply:
done